kyuubi-common/src/main/scala/org/apache/kyuubi/engine/deploy/yarn/EngineYarnModeSubmitter.scala

```diff
@@ -91,9 +98,36 @@ abstract class EngineYarnModeSubmitter extends Logging {
    */
   def engineExtraJars(): Seq[File] = Seq.empty
 
-  protected def submitApplication(): Unit = {
+  def run(): Unit = {
     yarnConf = KyuubiHadoopUtils.newYarnConfiguration(kyuubiConf)
     hadoopConf = KyuubiHadoopUtils.newHadoopConf(kyuubiConf)
+    appUser = kyuubiConf.getOption(KYUUBI_SESSION_USER_KEY).orNull
```

Comment:
is this correct for GROUP and SERVER share level?

Reply:
`KYUUBI_SESSION_USER_KEY`  is set to `EngineRef#appUser` when submit Engines. So it's correct for all share levels.

Reply:
It's weird. I remember we use this key to distinguish the session user and app user in some cases ...

Reply:
SGTM